bmq-sdk/src/main/java/com/bloomberg/bmq/Subscription.java

```diff
@@ -0,0 +1,312 @@
+        sb.append("[ maxUnconfirmedMessages: ")
+                .append(getMaxUnconfirmedMessages())
+                .append(", maxUnconfirmedBytes: ")
+                .append(getMaxUnconfirmedBytes())
+                .append(", consumerPriority: ")
+                .append(getConsumerPriority())
+                .append(" ]");
+        return sb.toString();
+    }
+
+    /** A helper class to create immutable {@code Subscription} with custom settings. */
+    public static class Builder {
+        private Optional<Long> maxUnconfirmedMessages;
+        private Optional<Long> maxUnconfirmedBytes;
+        private Optional<Integer> consumerPriority;
+        private SubscriptionExpression.Builder expressionBuilder;
+
+        /**
+         * Returns subscription parameters based on this object properties.
+         *
+         * @return Subscription immutable subscription parameters object
+         */
+        public Subscription build() {
+            return new Subscription(this);
+        }
+
+        /**
+         * Sets maximum number of outstanding messages that can be sent by the broker without being
+         * confirmed for this subscription.
+         *
+         * @param maxUnconfirmedMessages maximum number of outstanding messages
+         * @return Builder this object
+         */
+        public Subscription.Builder setMaxUnconfirmedMessages(long maxUnconfirmedMessages) {
+            this.maxUnconfirmedMessages = Optional.of(maxUnconfirmedMessages);
+            return this;
+        }
+
+        /**
+         * Sets maximum accumulated bytes of all outstanding messages that can be sent by the broker
+         * without being confirmed for this subscription.
+         *
+         * @param maxUnconfirmedBytes maximum accumulated bytes of all outstanding messages without
+         *     being confirmed
+         * @return Builder this object
+         */
+        public Subscription.Builder setMaxUnconfirmedBytes(long maxUnconfirmedBytes) {
+            this.maxUnconfirmedBytes = Optional.of(maxUnconfirmedBytes);
+            return this;
+        }
+
+        /**
+         * Sets priority of a consumer for this subscription with respect to delivery of messages.
+         *
+         * @param consumerPriority priority of a consumer
+         * @return Builder this object
+         */
+        public Subscription.Builder setConsumerPriority(int consumerPriority) {
+            this.consumerPriority = Optional.of(consumerPriority);
+            return this;
+        }
+
+        /**
+         * Sets expression text for this subscription.
+         *
+         * @param expression expression text
+         * @return Builder this object
+         */
+        public Subscription.Builder setExpressionText(String expression) {
+            this.expressionBuilder.setExpression(expression);
+            if (expression.length() > 0) {
+                setExpressionVersion(SubscriptionExpression.Version.e_VERSION_1);
```

Comment:
I hid the expression text setter here to simplify usage. Not sure if expr length check is needed though.